### ISSUE TITLE
Can't Manage to compile server_traits example

### ISSUE THREAD

**Issue by Enderh5** [oxalica/async-lsp#10]
I've been trying to compile the server_traits example but I'm getting the following error:
```Rust
error[E0277]: the trait bound `PipeStdin: futures::AsyncRead` is not satisfied
   --> src/main.rs:140:25
    |
140 |     server.run_buffered(stdin, stdout).await.unwrap();
    |            ------------ ^^^^^ the trait `futures::AsyncRead` is not implemented for `PipeStdin`
    |            |
    |            required by a bound introduced by this call
    |
    = help: the following other types implement trait `futures::AsyncRead`:
              Box<T>
              futures::io::Empty
              futures::io::Repeat
              Either<A, B>
              IntoAsyncRead<St>
              AllowStdIo<T>
              futures::io::BufReader<R>
              futures::io::BufWriter<W>
            and 7 others
note: required by a bound in `MainLoop::<S>::run_buffered`
   --> /home/rodrigo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-lsp-0.2.0/src/lib.rs:521:49
    |
521 |     pub async fn run_buffered(self, input: impl AsyncRead, output: impl AsyncWrite) -> Result<()> {
    |                                                 ^^^^^^^^^ required by this bound in `MainLoop::<S>::run_buffered`

error[E0277]: the trait bound `PipeStdout: futures::AsyncWrite` is not satisfied
   --> src/main.rs:140:32
    |
140 |     server.run_buffered(stdin, stdout).await.unwrap();
    |            ------------        ^^^^^^ the trait `futures::AsyncWrite` is not implemented for `PipeStdout`
    |            |
    |            required by a bound introduced by this call
    |
    = help: the following other types implement trait `futures::AsyncWrite`:
              Box<T>
              futures::io::Sink
              Vec<u8>
              Either<A, B>
              IntoAsyncRead<St>
              AllowStdIo<T>
              futures::io::BufReader<R>
              futures::io::BufWriter<W>
            and 8 others
note: required by a bound in `MainLoop::<S>::run_buffered`
   --> /home/rodrigo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/async-lsp-0.2.0/src/lib.rs:521:73
    |
521 |     pub async fn run_buffered(self, input: impl AsyncRead, output: impl AsyncWrite) -> Result<()> {
    |                                                                         ^^^^^^^^^^ required by this bound in `MainLoop::<S>::run_buffered`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `natural-language-lsp` (bin "natural-language-lsp") due to 2 previous errors
```

Do I have to install an specific version of any plugin?
Or disable a feature?

Either way this should probably be specified somewhere in the example.

**Comment by oxalica** [comment]
> I've been trying to compile the server_traits example

It [passes the CI](https://github.com/oxalica/async-lsp/actions/runs/7436529011/job/20233093473#step:8:112) and at least builds fine for me with `cargo build --all-features --example=server_trait` using `rustc 1.76.0`.

The message shows that you are building your own crate `natural-language-lsp`. Could you provide the source of it? It seems that you missed the call `PipeStdin::lock_tokio()` (for `tokio`) or `async_io::Async::new()` (for `async_io`), but instead directly plugging `PipeStdin` into `run_buffered`. The piped IO must have an asynchronous reactor/executor attached to function and cannot work on their own.

**Comment by Enderh5** [comment]
Sorry, I didn't explain myself correctly.
I was trying to build a small lsp and i tried to build it based on the example.
The code was just the same as the example.

I have managed to get it to build, I got the versions from this projects `cargo.toml` (Yes, dumb mistake sorry I'm new to this kind of projects in rust) and I had to change the `tokio_lock()` method to `lock()`.

Sorry for the inconvenience
